src/execution/execute.js

```diff
@@ -977,6 +1022,47 @@ function completeAbstractValue(
 }
 
 /**
+ * If a resolveType function is not given, then a default resolve behavior is
+ * used which tests each possible type for the abstract type by calling
+ * isTypeOf for the object being coerced, returning the first type that matches.
+ */
+function resolveFirstValidType(
```

Comment:
Suggestion: move this to the bottom of the file for less entropy in the diff.

Also, instead of having a default value for arguments, you could keep the `defaultResolveTypeFn` function, but just have it collect the possibleTypes and then call this function.